src/pages/workspaces/workspace/useWorkspace.test.ts

```diff
@@ -0,0 +1,457 @@
+      createdDate: '2023-02-03T22:26:06.124Z',
+      googleProject: 'google-project-id',
+      isLocked: false,
+      lastModified: '2023-02-03T22:26:06.202Z',
+      name: 'testName',
+      namespace: 'testNamespace',
+      workspaceId: 'google-workspace-id',
+      workspaceType: 'rawls',
+      workspaceVersion: 'v2'
+    },
+    canShare: true,
+    canCompute: true,
+    workspaceInitialized: true
+  }
+
+  const initializedAzureWorkspace = {
+    accessLevel: 'PROJECT_OWNER',
+    owners: [
+      'christina@foo.com'
+    ],
+    azureContext: {
+      managedResourceGroupId: 'test-mrg',
+      subscriptionId: 'test-sub-id',
+      tenantId: 'test-tenant-id'
+    },
+    workspace: {
+      attributes: {
+        description: ''
+      },
+      authorizationDomain: [],
+      bucketName: '',
+      cloudPlatform: 'Azure',
+      completedCloneWorkspaceFileTransfer: '2023-02-03T22:29:04.319Z',
+      createdBy: 'christina@foo.com',
+      createdDate: '2023-02-03T22:26:06.124Z',
+      googleProject: '',
+      isLocked: false,
+      lastModified: '2023-02-03T22:26:06.202Z',
+      name: 'testName',
+      namespace: 'testNamespace',
+      workspaceId: 'azure-workspace-id',
+      workspaceType: 'rawls',
+      workspaceVersion: 'v2'
+    },
+    canShare: true,
+    canCompute: true,
+    workspaceInitialized: true
+  }
+
+  const bucketLocationResponse = {
+    location: 'bucket-location',
+    locationType: 'location-type'
+  }
+
+  const defaultGoogleBucketOptions = {
+    googleBucketLocation: defaultLocation,
+    googleBucketType: locationTypes.default
+  }
+
+  const azureStorageDetails = {
+    location: 'container-location',
+    sas: { url: 'container-url?sas-token' }
+  }
+
+  const defaultAzureStorageOptions = {
+    azureContainerRegion: undefined,
+    azureContainerUrl: undefined,
+    azureContainerSasUrl: undefined
+  }
+
+  beforeEach(() => {
+    workspaceStore.reset()
+    jest.useFakeTimers()
+
+    // @ts-ignore
+    getUser.mockReturnValue({
+      email: 'christina@foo.com'
+    })
+
+    jest.spyOn(workspaceStore, 'set')
+    jest.spyOn(WorkspaceUtils, 'updateRecentlyViewedWorkspaces')
+    jest.spyOn(GoogleStorage, 'saToken')
+    jest.spyOn(Notifications, 'notify')
+
+    // Don't show expected error responses in logs
+    jest.spyOn(console, 'error').mockImplementation(() => {})
+    jest.spyOn(console, 'log').mockImplementation(() => {})
+  })
+
+  afterAll(() => {
+    jest.useRealTimers()
+  })
+
+  const assertResult = (result, expectedWorkspace, expectedStorageDetails, expectedAccessError) => {
+    expect(result.workspace).toEqual(expectedWorkspace)
+    expect(result.storageDetails).toEqual(expectedStorageDetails)
+    expect(result.accessError).toBe(expectedAccessError)
+    expect(result.refreshWorkspace).toBeTruthy()
+    expect(result.loadingWorkspace).toBe(false)
+  }
+
+  it('can initialize from a Google workspace in workspaceStore', async () => {
+    // Arrange
+    // @ts-ignore
```

Comment:
@msilva-broad can you help me figure out how to avoid tests ignores related to mocking Ajax? I looked at other examples in tests, but I don't understand the PartialContract syntax and I couldn't figure out something that worked.